server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -835,11 +835,12 @@ private void updatePayload(final Handle handle, final DataSegment segment) throw
 
   public List<DataSegment> getUnusedSegmentsForInterval(final String dataSource, final Interval interval)
```

Comment:
There's a lot of overhead in the lifecycle of the segment manager. Instead of having all the peons try and figure out what their view of all the segments should be, I instead elected to move the readonly transaction method into the connector.